modules/unsupported/geometry/src/main/java/org/geotools/geometry/iso/root/GeometryImpl.java

```diff
@@ -275,6 +276,15 @@ public final double getDistance(Geometry geometry) {
 	 * @see org.opengis.geometry.coordinate.root.Geometry#distance(org.opengis.geometry.coordinate.root.Geometry)
 	 */
 	public final double distance(Geometry geometry) {
+	        /* for 3D coordinate geometry */
+	        int g0Dim = getCoordinateDimension();
+	        int g1Dim = geometry.getCoordinateDimension();
+	        
+	        if(g0Dim == 3 && g1Dim == 3) {
+	            return Geometry3DOperation.distance(this, geometry);
+	        }
+	        /* */
```

Comment:
If the dimensions of geometries are different, how will you handle the case? For example, g0Dim == 3 && g1Dim == 2.